astropy/modeling/fitting.py

```diff
@@ -549,24 +549,32 @@ def __call__(self, model, x, y, z=None, weights=None, **kwargs):
         fitted_model = self.fitter(model, x, y, z, weights=weights, **kwargs)
         if z is None:
             filtered_data = y
+            filtered_weights = weights
             for n in range(self.niter):
                 filtered_data = self.outlier_func(filtered_data - fitted_model(x),
                                                   **self.outlier_kwargs)
                 filtered_data += fitted_model(x)
+                if weights is not None:
```

Comment:
No need to have this check. It happens inside the fitter instance.

Reply:
If weights is not None, I should filter it by [~filtered_data.mask]
If weights is None, i should not try to filter it by [~filtered_data.mask]

That's why there is that condition, for attempting to filter or not